### ISSUE TITLE
RUF026 does not support t-strings

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#19993]
### Summary

[`default-factory-kwarg` (RUF026)](https://docs.astral.sh/ruff/rules/default-factory-kwarg/) does not recognize t-strings as never being callable, as it does normal strings, causing false positives. [Example](https://play.ruff.rs/d168eab0-ac35-478a-bf7a-d3bd1bb375cc):
```console
$ cat >ruf026.py <<'# EOF'
from collections import defaultdict
defaultdict(default_factory=t"")
defaultdict(default_factory="")
# EOF

$ ruff --isolated check ruf026.py --select RUF026 --target-version py314 --preview --unsafe-fixes --fix
Found 1 error (1 fixed, 0 remaining).

$ cat ruf026.py 
from collections import defaultdict
defaultdict(t"")
defaultdict(default_factory="")
```

### Version

ruff 0.12.9 (ef422460d 2025-08-14)
